DFC.Digital/DFC.Digital.Web.Sitefinity.Core/DataEventHandling/MicroServicesPublishingPageBuilder.cs

```diff
@@ -34,44 +34,74 @@ public static IList<string> GetPageURLs(PageNode pageNode)
+
+            var pageDraftControls = pageDraft.Controls.Where(c => c.Caption == Digital.Core.Constants.ContentBlock);
 
-            //This bit came from Sitefinity Support.
-            var pageDraftControls = pageData.Controls.Where(c => c.Caption == Digital.Core.Constants.ContentBlock);
             foreach (var pageDraftControl in pageDraftControls)
             {
-                contentData += sitefinityManagerProxy.GetControlContent(providerName, pageDraftControl);
+                microServicesPublishingPageData.Content += sitefinityManagerProxy.GetControlContent(pageDraftControl);
             }
 
-            return contentData;
+            return microServicesPublishingPageData;
         }
 
-        public string GetMicroServiceEndPointConfigKeyForPageNode(string providerName, Type contentType, Guid itemId)
+        private MicroServicesPublishingPageData BuildPageData(PageNode pageNode, PageData pageData,  string providerName = null)
         {
-            var pageNode = sitefinityManagerProxy.GetPageNode(providerName, contentType, itemId);
+            var microServicesPublishingPageData = BuildBasePageData(pageNode, pageData);
 
-            //If the custom field is not there this will throw a system exception
-            //as we dont want to catch unnecessary exceptions this custom page field should be created.
-            return sitefinityPageNodeProxy.GetCustomField(pageNode, Digital.Core.Constants.MicroServiceEndPointConfigKey)?.Trim();
+            var pageControls = pageData.Controls.Where(c => c.Caption == Digital.Core.Constants.ContentBlock);
+            foreach (var pageControl in pageControls)
+            {
+                microServicesPublishingPageData.Content += sitefinityManagerProxy.GetControlContent(pageControl, providerName);
+            }
+
+            return microServicesPublishingPageData;
+        }
+
+        private MicroServicesPublishingPageData BuildBasePageData(PageNode pageNode, PageData pageData)
+        {
+            var microServicesPublishingPageData = new MicroServicesPublishingPageData() { CanonicalName = sitefinityPageNodeProxy.GetPageName(pageNode).ToLower() };
+            microServicesPublishingPageData.IncludeInSiteMap = pageNode.Crawlable;
+            microServicesPublishingPageData.AlternativeNames = GetPageURLs(pageNode);
+            microServicesPublishingPageData.LastReviewed = sitefinityPageNodeProxy.GetLastPublishedDate(pageNode);
+            microServicesPublishingPageData.Id = pageNode.Id;
+
+            microServicesPublishingPageData.BreadcrumbTitle = sitefinityPageDataProxy.GetTitle(pageData);
+            microServicesPublishingPageData.MetaTags = new MetaTags() { Description = sitefinityPageDataProxy.GetDescription(pageData), Keywords = sitefinityPageDataProxy.GetKeywords(pageData), Title = sitefinityPageDataProxy.GetHtmlTitle(pageData) };
+            return microServicesPublishingPageData;
```

Comment:
Minor point - could you initialise all this objects properties in the same way as you're doing with CanonicalName? Also you could then just return that rather than creating a variable to return